lusid-sdk-python/tests/test_instrument_master.py

```diff
@@ -0,0 +1,230 @@
+                                                            properties=[inst_isin, inst_sedol])
+
+        # upsert_response = cls.client.upsert_instruments({"correlationId3": instrument_definition})
+
+        inst_isin = models.InstrumentProperty(cls.ISIN_PROPERTY_KEY, models.PropertyValue("GB00B019KW72"))
+        inst_sedol = models.InstrumentProperty(cls.SEDOL_PROPERTY_KEY, models.PropertyValue("B019KW7"))
+
+        instrument_definition4 = models.InstrumentDefinition(name="SAINSBURY (J) PLC",
+                                                            identifiers={cls.FIGI_SCHEME: "BBG000BF0KW3",
+                                                                         cls.CUSTOM_INTERNAL_SCHEME: "INTERNAL_ID_4"},
+                                                            properties=[inst_isin, inst_sedol])
+
+        # upsert_response = cls.client.upsert_instruments({"correlationId4": instrument_definition})
+
+        inst_isin = models.InstrumentProperty(cls.ISIN_PROPERTY_KEY, models.PropertyValue("GB0008782301"))
+        inst_sedol = models.InstrumentProperty(cls.SEDOL_PROPERTY_KEY, models.PropertyValue("0878230"))
+
+        instrument_definition5 = models.InstrumentDefinition(name="TAYLOR WIMPEY PLC",
+                                                            identifiers={cls.FIGI_SCHEME: "BBG000BF4KL1",
+                                                                         cls.CUSTOM_INTERNAL_SCHEME: "INTERNAL_ID_5"},
+                                                            properties=[inst_isin, inst_sedol])
+
+        upsert_response = cls.client.upsert_instruments(
+            {
+                "correlationId1": instrument_definition1,
+                "correlationId2": instrument_definition2,
+                "correlationId3": instrument_definition3,
+                "correlationId4": instrument_definition4,
+                "correlationId5": instrument_definition5
+            }
+        )
+
+        assert len(upsert_response.values) == 5
+
+
+    def test_lookup_instrument_by_unique_id(self):
+
+        # Look up an instrument that already exists in the instrument master by a
+        # unique id, in this case an OpenFigi, and also return a list of aliases
+
+        fbn_ids = self.client.get_instruments(self.FIGI_SCHEME, ["BBG000C6K6G9", "BBG000BF4KL1"],
+                                              instrument_property_keys=[self.ISIN_PROPERTY_KEY,
+                                                                        self.SEDOL_PROPERTY_KEY])
+
+        self.assertEqual(fbn_ids.values["BBG000C6K6G9"].name, "VODAFONE GROUP PLC")
+
+        # get instrument from the master
+        fbn_inst = fbn_ids.values["BBG000C6K6G9"]
+        #fbn_inst = self.client.get_instrument(self.FIGI_SCHEME, "BBG000C6K6G9")
+        assert fbn_inst.name == "VODAFONE GROUP PLC"
+
+        # get isin
+        assert fbn_ids.values["BBG000C6K6G9"].properties[0].key == self.ISIN_PROPERTY_KEY
```

Comment:
check with Dan if the order of the properties is guaranteed to be same as when they were created, if not you will need to sort them before doing the asserts

Reply:
they are not guaranteed so I rewrote